### ISSUE TITLE
[Bug]: Accessibility violations

### ISSUE THREAD

**Issue by zarahzachz** [storybookjs/storybook#22504]
### Describe the bug

Just wanted to mention - I just bumped a project I'm working on to Storybook 7 and I'm getting a lot of Axe accessibility violations. Most seem centered around color contrast (the document section labels have low contrast) and form `id`s not being unique. 

I checked the Create React App (TypeScript) base project from [Storybook.new](https://new-storybook.netlify.app) to make sure this wasn't something my team introduced and I'm getting the same accessibility violations there. 

It'd be great if Storybook were accessible out of the box.

### To Reproduce

You can find the accessibility bugs in the base Create React App (TypeScript) project via storybook.new. If you open this project in its own window and run an audit using an accessibility tool like Axe, you'll find errors around color contrast, unique `id`s, etc.

### System

```shell
I'm using MacOS + Chrome.
```

**Comment by Sidnioulz** [comment]
Hey Zara, thanks for the report!

We have a few dozen reported accessibility issues, many of which have been worked on and many more of which have a work plan established with accessibility specialists. We're going through the backlog, and always appreciate contributions!

As this issue is fairly generic and does not track any work item in particular, I'll take the liberty to close it. We prefer to have issues scoped to a specific problem / axe violation so that it gives contributors a clearer goal on what to do to help. Thanks again!
